src/performance/utils.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { Profiler, useCallback, useRef } from 'react';
+
+const events = new Map<string, number>();
+
+export function measureEventStart(name: string) {
+  events.set(name, global.performance.now());
+}
+
+export function measureEventEnd(name: string) {
+  const total = global.performance.now() - events.get(name)!;
+  events.set(name, total);
+  global.console.log(`Event ${name}: ${total.toFixed(2)}ms\n`);
+}
+
+const timeoutRegistry = new Map<string, any>();
+
+function createDebounceAverage(
+  title: string,
+  phase: string,
+  delay: number = 100
+) {
+  let values: number[] = [];
+  let _timeout: NodeJS.Timeout;
```

Comment:
```suggestion
  let timeout: NodeJS.Timeout;
```

I'd keep the name simpler and just go with `_timeout`.